shared-libs/contacts/src/places.js

```diff
@@ -83,23 +83,29 @@ const validatePlace = place => {
   return Promise.resolve();
 };
 
-const createPlace = place => {
-  const self = module.exports;
-  return self._validatePlace(place)
-    .then(() => {
-      const date = place.reported_date ? utils.parseDate(place.reported_date) : new Date();
-      place.reported_date = date.valueOf();
-      if (place.parent) {
-        place.parent = lineage.minifyLineage(place.parent);
-      }
-      if (place.contact) {
-        // also validates contact if creating
-        return people.getOrCreatePerson(place.contact).then(person => {
-          place.contact = lineage.minifyLineage(person);
-        });
-      }
-    })
-    .then(() => db.medic.post(place));
+const createPlace = async (place) => {
+  const contact = place.contact;
+  delete place.contact;
+
+  await module.exports._validatePlace(place);
+  people._validatePerson(contact);
```

Comment:
I believe that this function will return an error instead of throwing an error. Like here: https://github.com/medic/cht-core/blob/017f395945e9fd1b9c6e5b31041aece7c344f6bf/shared-libs/contacts/src/people.js#L68

So we'll need to treat the error similarly as in the example I linked.